vllm/model_executor/layers/linear.py

```diff
@@ -403,6 +440,13 @@ def weight_loader(self,
             shard_size = loaded_weight.shape[0]
             shard_offset = loaded_shard_id * shard_size
             param_data = param_data.narrow(0, shard_offset, shard_size)
+
+        # If a param_shard_splitter is defined by the LinearMethod, use it.
```

Comment:
I'm assuming we can get rid of all of these after refactoring weight loading, but meanwhile I'd prefer a more detail comment to explain what `param_shard_splitter` does here.

Reply:
This does the same thing as `scale_shard_splitter` we had for fp8 ... we can rename to match fp8

but yes this will be addressed by the refactor